public/js/lib/spell-checker/spell-checker.js

```diff
@@ -0,0 +1,389 @@
+          return `${BASE_STYLE_CSS_CLASS}-error`
+        }
+
+        return `${BASE_STYLE_CSS_CLASS}-warning`
+      },
+      startState: function () {
+        return { lineCount: 0, charCount: 0, match: null }
+      },
+      blankLine: function (state) {
+        state.lineCount++
+      }
+    }
+
+    const mode = codeMirrorInstance.getMode(
+      config, config.backdrop || 'text/plain'
+    )
+
+    return codeMirrorInstance.overlayMode(mode, overlay, true)
+  })
+}
+
+SpellChecker.data = null
+SpellChecker._overlay = null
+SpellChecker._openMatch = null
+SpellChecker.currentRequest = null
+
+/**
+ * Fetches data from the LanguageTool HTTP server via the specified endpoint
+ * to check the content of the editor.
+ *
+ * @param {object} editor - The CodeMirror editor instance.
+ */
+SpellChecker.fetchData = (editor) => {
+  SpellChecker.currentRequest = $.post(`${serverurl}/check/`, {
+    text: editor.getValue(),
+    language: 'auto',
+    motherTongue: 'fr'
+  })
+    .done(data => {
+      // LanguageTool returns an offset, but CodeMirror needs a line and character position
+      data.matches = data.matches.map((match) => {
+        // Convert global offset to line and character position
+        return Object.assign({}, match, {
+          position: editor.posFromIndex(match.offset)
+        })
+      })
+      if (debug) {
+        console.debug(data)
+      }
+      SpellChecker.data = data
+      SpellChecker.render(editor)
+    })
+    .fail((err) => {
+      if (debug) {
+        console.debug(err)
+      }
+    })
+}
+
+SpellChecker.render = (editor) => {
+  editor.setOption('mode', config.spellCheckerMode)
+}
+
+SpellChecker.hasError = (token) => {
+  return token && token.state && token.state.overlayCur && token.state.overlayCur.includes(BASE_STYLE_CSS_CLASS)
+}
+
+SpellChecker.getOverlayPosition = (cursorPosition) => {
+  const bottomSpace = window.innerHeight - cursorPosition.bottom
+  const rightSpace = window.innerWidth - cursorPosition.right
+
+  // By default, the overlay is positioned at the bottom-right of the cursor
+  let top = cursorPosition.top
+  let left = cursorPosition.left
+
+  // FIXME: hardcoded values
+  const overlay = {
+    width: 300,
+    height: 162
+  }
+
+  // If there is limited space at the bottom, position the overlay on the top
+  if (bottomSpace < overlay.height) {
+    top -= overlay.height + 10
+  } else {
+    // FIXME: arbitrary hardcoded value to open overlay slightly under the cursor
+    top += 30 // Assuming a standard lineheight
+  }
+
+  // If there is limited space at the right, position the overlay on the left side
+  if (rightSpace < overlay.width) {
+    left -= overlay.width
+  }
+
+  return { top, left }
+}
+
+/**
+ * This function uses the close icon from Bootstrap.
+ * @see {@link https://icons.getbootstrap.com/icons/x/}
+ * @returns {string} HTML string for the close button.
+ */
+SpellChecker.createHtmlCloseButton = () => {
+  return `
+    <button id="close-overlay">
```

Comment:
```suggestion
    <button id="close-overlay" type="button">
```

Not a _necessary_ change I know but here we make sure the button won't ever submit a potential form it's included in by mistake.